### ISSUE TITLE
custom_federated_algorithms_2 + collab = error

### ISSUE THREAD

**Issue by pnzr00t** [tensorflow/federated#800]
Hello. Try run example in collab and get error.

Collab info 
```
import sys
print("Python version")
print (sys.version)
print("Version info.")
print (sys.version_info)
print(tf.__version__)
print(tff.version)
```

> Python version
> 3.6.9 (default, Nov  7 2019, 10:44:02) 
> [GCC 8.3.0]
> Version info.
> sys.version_info(major=3, minor=6, micro=9, releaselevel='final', serial=0)
> 2.1.0
> <module 'tensorflow_federated.version' from '/usr/local/lib/python3.6/dist-packages/tensorflow_federated/version.py'>

Error block:

```
initial_model = {
    'weights': np.zeros([784, 10], dtype=np.float32),
    'bias': np.zeros([10], dtype=np.float32)
}

sample_batch = federated_train_data[5][-1]

batch_loss(initial_model, sample_batch)
```
Error message:

> ---------------------------------------------------------------------------
> TypeError                                 Traceback (most recent call last)
> <ipython-input-26-6e50123e60ce> in <module>()
>       6 sample_batch = federated_train_data[5][-1]
>       7 
> ----> 8 batch_loss(initial_model, sample_batch)
> 
> 27 frames
> /usr/local/lib/python3.6/dist-packages/tensorflow_federated/python/core/impl/executors/eager_executor.py in to_representation_for_type(value, type_spec, device)
>     250       raise TypeError(
>     251           'The apparent type {} of a tensor {} does not match the expected '
> --> 252           'type {}.'.format(value_type, value, type_spec))
>     253     return value
>     254   elif isinstance(type_spec, computation_types.SequenceType):
> 
> TypeError: The apparent type float32[784,10] of a tensor [[0. 0. 0. ... 0. 0. 0.]
>  [0. 0. 0. ... 0. 0. 0.]
>  [0. 0. 0. ... 0. 0. 0.]
>  ...
>  [0. 0. 0. ... 0. 0. 0.]
>  [0. 0. 0. ... 0. 0. 0.]
>  [0. 0. 0. ... 0. 0. 0.]] does not match the expected type float32[10].

Printing "initial_model"

> {'bias': array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0.], dtype=float32),
>  'weights': array([[0., 0., 0., ..., 0., 0., 0.],
>         [0., 0., 0., ..., 0., 0., 0.],
>         [0., 0., 0., ..., 0., 0., 0.],
>         ...,
>         [0., 0., 0., ..., 0., 0., 0.],
>         [0., 0., 0., ..., 0., 0., 0.],
>         [0., 0., 0., ..., 0., 0., 0.]], dtype=float32)}

As i understood "initial_model" - created as sort dict, but it must be created as unsorted dict.

**Comment by pnzr00t** [comment]
If replace this code
```
initial_model = {
    'weights': np.zeros([784, 10], dtype=np.float32),
    'bias': np.zeros([10], dtype=np.float32)
}
```

To that 
```
initial_model = collections.OrderedDict([('weights', np.zeros([784, 10], dtype=np.float32)), 
                                          ('bias', np.zeros([10], dtype=np.float32))])
```
All works great

If my answer wrong i can only say sorry, i'm noob in python

**Comment by ZacharyGarrett** [comment]
@pnzr00t You're absolutely right, this needs to be changed. On `master` this has already been updated: https://github.com/tensorflow/federated/blob/master/docs/tutorials/custom_federated_algorithms_2.ipynb

However we need to update the tensorflow.org/federated site. I'll get on this.

**Comment by pnzr00t** [comment]
Yes sorry, just find it about 3 minutes ago, sorry for junk topic.

I study course in coursera, and after first part demonstration, i go to link notebook, and probably it was lin at  0.10 version. 

Any way, thank you for fast answer

**Comment by pnzr00t** [comment]
> @pnzr00t You're absolutely right, this needs to be changed. On `master` this has already been updated: https://github.com/tensorflow/federated/blob/master/docs/tutorials/custom_federated_algorithms_2.ipynb
> 
> However we need to update the tensorflow.org/federated site. I'll get on this.

I test 1 more time and found "my problem", in you "master" branch in notebook document, you have links  "View on TensorFlow.org | Run in Google Colab | View source on GitHub"

Links with "Run in Google Colab | View source on GitHub" is linked to URL https://colab.research.google.com/github/tensorflow/federated/blob/v0.12.0/docs/tutorials/custom_federated_algorithms_2.ipynb

Witch linked to 
> blob/v0.12.0/

**Comment by cramertj** [comment]
@pnzr00t Yes, the situation around docs versioning is unfortunate at the moment, and we're working to get this resolved internally. Thank you for your patience, and sorry for the confusion!
